src/shared/fdset.c

```diff
@@ -286,26 +330,27 @@ int fdset_close_others(FDSet *fds) {
 }
 
 unsigned fdset_size(FDSet *fds) {
-        return set_size(MAKE_SET(fds));
+        return hashmap_size(MAKE_HASHMAP(fds));
```

Comment:
hmm this one is weird, because it will return the number of un-indexed fds + 2x the number of indexed fds. I see no easy way to return something better, and given we will not mix entries of the two kinds probably OK to leave this way, but this requies a comment explaining this.